### DISCUSSION TITLE
独立笔记无法导出，大家帮我看看导出模板哪里出错了？

### DISCUSSION THREAD

**Discussion by XoYaPeng** [windingwind/zotero-better-notes#1168] [Q&A]
![image](https://github.com/user-attachments/assets/2a1efb88-acc2-4f0a-ab78-43f2a048562d)
这四种方式都不能导出为MD文件，更别说同步到OB里了哈哈哈
导出模板如下：
Export MD File Name (导出MD文件名)：`${(noteItem.getNoteTitle ? noteItem.getNoteTitle().replace(/[/\\?%*:|"<> ]/g, "-") : "")}_${noteItem.key}.md`

Export MD File Header (导出MD头)：`${await new Promise(async (r) => {
let header = {};
header.Title = noteItem.parentItem
 ? noteItem.parentItem.getField("title")
: "";
header.Journal = noteItem.parentItem? 
noteItem.parentItem.getField("publicationTitle") : "";
header.author = noteItem.parentItem
? noteItem.parentItem.getCreators().map((au) => au.firstName + " " + au.lastName).join("; ")
: "";
header.url = noteItem.parentItem? 
noteItem.parentItem.getField("url") : "";
header.DOI = noteItem.parentItem? 
noteItem.parentItem.getField("DOI") : "";
header.year = noteItem.parentItem.getField("date") 
? new Date(noteItem.parentItem.getField("date")).getFullYear() 
: '';
header.$date_created = noteItem.parentItem.getdate_created = new Date().toISOString().split('T')[0];
header.$shortTitle = noteItem.parentItem? 
noteItem.parentItem.getField("shortTitle") : "";
header.tags = noteItem.parentItem.getTags().map((_t) => _t.tag);
header.noteLink = Zotero.BetterNotes.api.convert.note2link(noteItem);
header.$collections = (
    await Zotero.Collections.getCollectionsContainingItems([
      (noteItem.parentItem || noteItem).id,
    ])
  ).map((c) => c.name);
r(JSON.stringify(header));
})}`

Export MD File Content (导出MD正文)：`${mdContent.replaceAll("\\[\\[", "[[").replaceAll("\\#", "#").replaceAll("> \\[!", "> [!").replaceAll("\\](.*)", "](.*)")}`

**Comment by windingwind** [comment]
请提供debug output

**Reply by XoYaPeng** [reply]
老大，是从这里看吗?
![image](https://github.com/user-attachments/assets/274d090c-bdbf-490a-969f-579619b8b841)

下面是复制的结果：
[JavaScript Error: "Style with ID http://www.zotero.org/styles/american-chemical-society already loaded from american-chemical-society.csl" {file: "chrome://zotero/content/xpcom/style.js" line: 182}]
Zotero.Styles</_readStylesFromDirectory<@chrome://zotero/content/xpcom/style.js:182:32
tryCatcher@resource://zotero/bluebird/util.js:16:23
module.exports/PromiseSpawn.prototype._promiseFulfilled@resource://zotero/bluebird/generators.js:97:49
module.exports/Promise.prototype._settlePromise@resource://zotero/bluebird/promise.js:609:26
module.exports/Promise.prototype._settlePromise0@resource://zotero/bluebird/promise.js:649:10
module.exports/Promise.prototype._settlePromises@resource://zotero/bluebird/promise.js:729:18
_drainQueueStep@resource://zotero/bluebird/async.js:93:12
_drainQueue@resource://zotero/bluebird/async.js:86:24
Async.prototype._drainQueues@resource://zotero/bluebird/async.js:102:16
Async/this.drainQueues@resource://zotero/bluebird/async.js:15:14


[JavaScript Error: "Attempt to override an existing message: "pdfReader-a11yTextualAnnotationInstruction"." {file: "reader.ftl" line: 0}]

[JavaScript Error: "Attempt to override an existing message: "pdfReader-a11yAnnotationInstruction"." {file: "reader.ftl" line: 0}]

[JavaScript Error: "Attempt to override an existing message: "pdfReader-a11yTextualAnnotationInstruction"." {file: "reader.ftl" line: 0}]

[JavaScript Error: "Attempt to override an existing message: "pdfReader-a11yAnnotationInstruction"." {file: "reader.ftl" line: 0}]

[JavaScript Error: "The connection was refused when attempting to contact wss://stream.zotero.org/."]

[JavaScript Error: "WebSocket connection closed: 1006 "]

[JavaScript Error: "TypeError: emptyListMessage is null" {file: "chrome://zotero/content/standalone/standalone.js" line: 687}]
ZoteroStandalone</this.updateAddonsPane/<@chrome://zotero/content/standalone/standalone.js:687:7


1729579654672	addons.xpi	WARN	Download of https://github.com/l0o0/jasminum/releases/latest/download/jasminum.xpi failed: 404 Not Found

[JavaScript Error: "TypeError: emptyListMessage is null" {file: "chrome://zotero/content/standalone/standalone.js" line: 687}]
ZoteroStandalone</this.updateAddonsPane/<@chrome://zotero/content/standalone/standalone.js:687:7


[JavaScript Error: "Passing a URI to Zotero.File.getContentsAsync() is deprecated -- use Zotero.HTTP.request() instead"]

[JavaScript Error: "Passing a URI to Zotero.File.getContentsAsync() is deprecated -- use Zotero.HTTP.request() instead"]

[JavaScript Error: "TypeError: this.editorView is undefined" {file: "resource://zotero/note-editor/editor.js" line: 6}]

[JavaScript Error: "TypeError: noteItem.parentItem is undefined" {file: "jar:file:///C:/Users/PengXY/AppData/Roaming/Zotero/Zotero/Profiles/w7fvuloo.zotero%207%20beta/extensions/Knowledge4Zotero@windingwind.com.xpi!/chrome/content/scripts/BetterNotes.js line 55462 > AsyncFunction" line: 17}]

appName => Zotero, version => 7.0.8 (x64), os => Windows 10.0 19045, locale => zh-CN, extensions => Zoplicate (3.0.3, extension), ZotCard (3.0.11, extension), Zotero Attanger (1.2.7, extension), Actions and Tags for Zotero (2.0.3, extension), Translate for Zotero (2.0.4, extension), Ethereal Reference (1.2.3, extension), Linter for Zotero (1.20.0, extension), Ethereal Style (5.1.7, extension), Awesome GPT (1.3.7, extension), Better Notes for Zotero (2.0.18, extension), Jasminum (1.0.0-16, extension, disabled), ZoteroExcalidraw (1.0.2, extension, disabled), PDF Figure (0.2.4, extension, disabled), Del Item With Attachment (0.3.3, extension, disabled), Easier Citation (0.4.4, extension, disabled), Zotero Pin Items (0.1.1, extension, disabled), Zotero Author Browser (0.1.4, extension, disabled), arXiv Workflow for Zotero (0.2.7, extension, disabled), Green Frog (0.19.13, extension, disabled), Zotero Categorial Tags (0.1.12, extension, disabled)

**Reply by windingwind** [reply]
header.year = noteItem.parentItem.getField("date")

这一行不正确，不能够假设parentItem存在

**Reply by XoYaPeng** [reply]
谢谢老大 我明天改一下
